isAnnualPlanOrUpgradeableAnnualPeriod: use feature checks and rename

mreishus opened the pull request:
#### Changes proposed in this Pull Request

* This is a tricky one! `isAnnualPlanOrUpgradeableAnnualPeriod` is a complex selector. Let's first start out with what it does. Ultimately, it controls some text about support on a marketplace plugins page. Two examples, where the text is in the bottom right:

**(Pro Plan Disabled In Code + Free Plan + Annual selected) = "Live Chat Support"**
![2022-05-11_17-15_1](https://user-images.githubusercontent.com/937354/167956509-27b34bf0-2cf4-4a0d-985c-58086006c9b5.png)

**(Pro Plan Disabled In Code + Free Plan + Monthly selected) = "Unlimited Chat Support"**
![2022-05-11_17-15](https://user-images.githubusercontent.com/937354/167956685-e028b73a-60af-49f7-be4e-bed0905f30b8.png)

* Notes about our current plans:
  * Most plans do provide Live Chat Support, but monthly plans do not.
  * The Pro Plan cannot be bought monthly, so, it always provides Live Chat Support.
  * The Pro Plan has a "master switch" and can be turned on or off for all of Calypso. Currently it's ON. To see any of these changes you will have to temporarily turn it OFF. If the PRO plan is on, the text always reads "Premium Support" and the selector is not even called here.

**(Pro Plan Enabled + Any Plan + Either Annual or Monthly Selected) = "Premium Support"**
![2022-05-11_17-21](https://user-images.githubusercontent.com/937354/167957100-89497464-7cd5-44df-90ad-f8d5c8242471.png)

* **What the selector used to do**
  *  Examine if you need to upgrade plan before purchasing marketplace addons (hardcoded list of plans).
    * If no upgrade is needed, return if the current plan you have is a yearly subscription.
      * Assumption: All yearly subscriptions have live support. 
    * If an upgrade is needed, return if you have selected "Annual" in the top right.
      * Assumption: The yearly subscription you are about to purchase has live support.
* **What the selector does now**
  *  Examine if you need to upgrade plan before purchasing marketplace addons (check install_plugins feature). 
    * If no upgrade is needed, return if you have the live_support feature.
    * If an upgrade is needed, return if you have selected "Annual" in the top right.
      * Assumption: The yearly subscription you are about to purchase has live support. 

I also renamed it `isAnnualPlanOrUpgradeableAnnualPeriod` -> `hasOrIntendsToBuyLiveSupport` to try to make its function clearer.

#### Testing instructions

* STOP! D80582-code must be merged or applied to your sandbox first.
* Visit `/plugins/woocommerce-subscriptions/<site>` across a variety of dimensions:
  * Pro plan **enabled** vs **disabled**
  * The site has a **free plan** vs a **pro plan**
  * In the top right you have clicked either **annual** or **monthly**
Everything should be exactly the same before and after.

![2022-05-11_17-31](https://user-images.githubusercontent.com/937354/167958150-a3fc6542-de85-4cfd-9832-85e6a5c5456f.png)

To disable PRO, apply this patch and run locally:
```diff
diff --git a/client/my-sites/plugins/plugin-details-CTA/usps.tsx b/client/my-sites/plugins/plugin-details-CTA/usps.tsx
index 3d61eca15e..825cdc0d9b 100644
--- a/client/my-sites/plugins/plugin-details-CTA/usps.tsx
+++ b/client/my-sites/plugins/plugin-details-CTA/usps.tsx
@@ -54,9 +54,7 @@ const USPS: React.FC< Props > = ( {
        const isAnnualPeriod = billingPeriod === IntervalLength.ANNUALLY;
 
        const selectedSite = useSelector( getSelectedSite );
-       const eligibleForProPlan = useSelector( ( state ) =>
-               isEligibleForProPlan( state, selectedSite?.ID )
-       );
+       const eligibleForProPlan = false;
 
        const planDisplayCost = useSelector( ( state ) => {
                const productSlug = isAnnualPeriod ? PLAN_BUSINESS : PLAN_BUSINESS_MONTHLY;
```

* Also code review - I have also changed the `shouldUpgradeCheck` function without checking the other places it's used in the UI.

#### History

#59853, #61212, #61365, #61371

Related to p4TIVU-a66-p2

mreishus committed:
Marketplace selectors: shouldUpgradeCheck and isAnnualPlanOrUpgradeableAnnualPeriod use feature checks

mreishus committed:
Rename isAnnualPlanOrUpgradeableAnnualPeriod -> hasOrIntendsToBuyLiveSupport

epeicher reviewed (approved):
Thank you very much for the detailed description!

I have checked all the combinations described and it works as advertised. 

I also have checked the usages of `shouldUpgradeCheck` selector and it works as expected:
![CleanShot 2022-05-12 at 13 51 21@2x](https://user-images.githubusercontent.com/3519124/168068926-25e753b6-5fd1-48dd-9661-02c2b18d63eb.png)

LGTM!

On an additional note, not directly related to this change. do you know if `Live chat support 24x7` should be listed in the **Support** section below? 
![CleanShot 2022-05-12 at 13 57 22@2x](https://user-images.githubusercontent.com/3519124/168069908-548a0f97-a1e6-4a8c-8f44-147b33467c3d.png)

Also, when proceeding to the checkout, should the `included with your purchase` have the same value than the **Support** section?
![CleanShot 2022-05-12 at 13 59 28@2x](https://user-images.githubusercontent.com/3519124/168070411-661634d1-2cdc-4b6c-ac6d-439ce10b2b76.png)


Just wanted to ask here but as mentioned, those questions are not directly related to this specific task.

mreishus committed:
Update plugin-details-sidebar to use the same support text as plugin-details-CTA

mreishus committed:
plugins-details-CTA, -sidebar: extract logic to new function usePluginsSupportText

mreishus commented:
> On an additional note, not directly related to this change. do you know if `Live chat support 24x7` should be listed in the **Support** section below?

Great catch. Both components here (`PluginDetailsSidebar`, `USPS`) and started with the same logic, but the pro-plan logic was added only to the `USPS` component. I also added it to the `PluginsDetailSidebar` logic to align the two. Then it seemed like a too-complex set of logic in two different places that could easily get out of sync again, so I extracted it into a new function `usePluginsSupportText()`.

epeicher reviewed (approved):
Thanks for the last change aligning `PluginDetailsSidebar` and `USPS` components and for the refactor, it makes total sense to me!

I have tested and it works as described.

LGTM 🚢 !
